src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -278,6 +284,87 @@ internal static object GetClaimValueUsingValueType(Claim claim)
             return claim.Value;
         }
 
+        /// <summary>
+        /// Gets ValueType of the claim from it's Value.
+        /// </summary>
+        /// <param name="value">Represents value of a claim.</param>
+        /// <returns>String representing claim's ValueType.</returns>
+        internal static string GetClaimValueTypeFromValue(object value)
+        {
+            if (value.GetType().Name == typeof(String).Name && value.ToString() == string.Empty)
```

Comment:
Can we use the `is` operator here instead of checking type via string comparison?

Reply:
Same below with checking for JObject and JArray

Reply:
Another option is to use `value.GetType().Equals(typeof(String))`.

Reply:
`value.GetType().Equals(typeof(String))`  option does not work for List. As list can contain object of any type.

1. `value.GetType()` resolves to `System.Collections.Generic.List`1[T]`
2. `typeof(List<>) `resolves to `System.Collections.Generic.List`1[System.<object type>]`